types/wordpress__blocks/index.d.ts

```diff
@@ -243,23 +245,26 @@ export interface BlockInstance<T extends Record<string, any> = { [k: string]: an
     readonly originalContent?: string | undefined;
 }
 
-export interface BlockDeprecation<T extends Record<string, any>>
-    extends Pick<Block<T>, 'attributes' | 'save' | 'supports'> {
+export interface BlockDeprecation<
+    // The new block attribute types.
+    N extends Record<string, any>,
+    // The old block attribute types.
+    O extends Record<string, any> = Record<string, any>,
+> extends Pick<Block<O>, 'attributes' | 'save' | 'supports'> {
     /**
      * A function which, given the attributes and inner blocks of the
      * parsed block, returns true if the deprecation can handle the block
      * migration. This is particularly useful in cases where a block is
      * technically valid even once deprecated, and requires updates to its
      * attributes or inner blocks.
      */
-    isEligible?(attributes: Record<string, any>, innerBlocks: BlockInstance[]): boolean;
+    isEligible?(attributes: O, innerBlocks: BlockInstance[]): boolean;
```

Comment:
I don't think this type is correct here because we could be dealing with attributes of a different shape than we expect, which is partially the purpose of the function. that is, `Record<string, any>` better captures that we aren't sure what shape of attributes we will receive, _but in the case that we do recognize them as `O`_ we will say, "yes, we can handle this migration." Otherwise it's our expectation that we might receive attributes that _aren't_ of type `O`.

Does that seem right to you?